pkg/nodebootstrap/userdata_al2.go

```diff
@@ -24,6 +24,11 @@ func makeAmazonLinux2Config(spec *api.ClusterConfig, ng *api.NodeGroup) (configF
 		return nil, err
 	}
 
+	maxPodsData, err := max_pods_mapTxtBytes()
```

Comment:
Can we make the `max_pods_map.txt` on provisioning time from the former `pkg/nodebootstrap/maxpods.go`, rather than go-generating it as a txt file?

The rationale is that I have a potential(very sorry but not raised yet) feature that would allow us to tune the "ratio" of max pods per instance type, e.g. `1/2 of default MAX_PODS`. Setting the uniform max pods via configs won't work for mixed instance types.

Leaving `pkg/nodebootstrap/maxpods.go` as-is allows both making this PR smaller, while the code required to implement the potential feature minimum.

WDYT?

Reply:
Done :) The only issue is that the file we are generating does not have a deterministic order. Maybe it would be good to sort the entries but not sure if it's important. @errordeveloper what do you think?